From 085e85e23d86b49cae114fcbe322325462c058cc Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Marek=20Barto=C5=A1?= <bartos.developer152@gmail.com>
Date: Fri, 23 Dec 2022 16:27:23 +0100
Subject: [PATCH] Fix coverage failure in case of an invalid file

---
 src/CodeCoverage.php | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/src/CodeCoverage.php b/src/CodeCoverage.php
index c35f27c8..9340f673 100644
--- a/src/CodeCoverage.php
+++ b/src/CodeCoverage.php
@@ -550,7 +550,11 @@ private function processUncoveredFilesFromFilter(): void
 
         foreach ($uncoveredFiles as $uncoveredFile) {
             if ($this->filter->isFile($uncoveredFile)) {
-                include_once $uncoveredFile;
+                try {
+                    include_once $uncoveredFile;
+                } catch (\Error $e) {
+                    // Include may fail if file content is not valid (due to implementing/extending non-existent class, ...)
+                }
             }
         }
 
-- 
2.34.1.windows.1

